### PR TITLE
[WIP] Deprec docs

### PR DESCRIPTION
**Pull request by waterponey**

### CODE REVIEW

File: sklearn/cross_validation.py [hunk 1]
```diff
@@ -266,6 +280,10 @@ def __init__(self, n, n_folds, shuffle, random_state):
 class KFold(_BaseKFold):
     """K-Folds cross validation iterator.
 
+    .. deprecated:: 0.18
+    This module will be removed in 0.20.
```
Comment (jnothman) [inline sklearn/cross_validation.py]: Indentation

File: sklearn/cross_validation.py [hunk 2]
```diff
@@ -1085,6 +1115,10 @@ def __len__(self):
 class PredefinedSplit(_PartitionIterator):
     """Predefined split cross validation iterator
 
+    .. deprecated:: 0.18
+    This module will be removed in 0.20.
```
Comment (jnothman) [inline sklearn/cross_validation.py]: Indentation

File: sklearn/cross_validation.py [hunk 3]
```diff
@@ -1241,6 +1275,10 @@ def cross_val_predict(estimator, X, y=None, cv=None, n_jobs=1,
                       verbose=0, fit_params=None, pre_dispatch='2*n_jobs'):
     """Generate cross-validated estimates for each input data point
 
+    .. deprecated:: 0.18
+    This module will be removed in 0.20.
```
Comment (jnothman) [inline sklearn/cross_validation.py]: Indentation

File: sklearn/datasets/lfw.py [hunk 4]
```diff
@@ -382,6 +382,10 @@ def _fetch_lfw_pairs(index_file_path, data_folder_path, slice_=None,
 def load_lfw_people(download_if_missing=False, **kwargs):
     """Alias for fetch_lfw_people(download_if_missing=False)
 
+    .. deprecated:: 0.17
+    Deprecated :func:`load_flw_people` will be removed in 0.19.
```
Comment (jnothman) [inline sklearn/datasets/lfw.py]: Indentation

File: sklearn/datasets/lfw.py [hunk 5]
```diff
@@ -511,6 +515,10 @@ def fetch_lfw_pairs(subset='train', data_home=None, funneled=True, resize=0.5,
 def load_lfw_pairs(download_if_missing=False, **kwargs):
     """Alias for fetch_lfw_pairs(download_if_missing=False)
 
+    .. deprecated:: 0.17
+    Deprecated :func:`load_flw_pairs` will be removed in 0.19.
```
Comment (jnothman) [inline sklearn/datasets/lfw.py]: Indentation

File: sklearn/decomposition/pca.py [hunk 6]
```diff
@@ -526,6 +526,11 @@ def score(self, X, y=None):
 class RandomizedPCA(BaseEstimator, TransformerMixin):
     """Principal component analysis (PCA) using randomized SVD
 
+    .. deprecated:: 0.18
+    Deprecated :class:`RandomizedPCA` will be removed in 0.20.
```
Comment (jnothman) [inline sklearn/decomposition/pca.py]: Indentation

File: sklearn/learning_curve.py [hunk 7]
```diff
@@ -32,6 +32,10 @@ def learning_curve(estimator, X, y, train_sizes=np.linspace(0.1, 1.0, 5),
                    error_score='raise'):
     """Learning curve.
 
+    .. deprecated:: 0.18
+    This module will be removed in 0.20.
```
Comment (jnothman) [inline sklearn/learning_curve.py]: Indentation

File: sklearn/cross_validation.py [hunk 8]
```diff
@@ -62,6 +62,12 @@
            'train_test_split']
 
 
+warnings.warn("This module was deprecated in version 0.18 in favor of the "
```
Comment (jnothman) [inline sklearn/cross_validation.py]: This is already above.

**Comment by jnothman** [comment]
You need to handle `LeaveOneLabelOut` and the other `*Label*` splitters, and map them to the new `model_selection.LeaveOneGroupOut` etc.

**Comment by jnothman** [comment]
But I can certainly see the "deprecated" messages in the rendered docs.

File: doc/modules/classes.rst [hunk 9]
```diff
@@ -315,7 +315,6 @@ Samples generator
    decomposition.PCA
    decomposition.IncrementalPCA
    decomposition.ProjectedGradientNMF
-   decomposition.RandomizedPCA
```
Comment (jnothman) [inline doc/modules/classes.rst]: this should reappear below

**Comment by jnothman** [comment]
And thanks!

**Comment by waterponey** [comment]
Yup for some reason when I do the make html-noplot on my computer it doesn't work , but on circleci it does...

I'll do the correction and add the mixin.

Also, for LDA and QDA that are just aliases, the deprecated message appears in the LinearDiscriminantAnalysis page and not the LDA page since I put it in this doc. But I think it's a bit confusing...

File: doc/modules/classes.rst [hunk 10]
```diff
@@ -1365,4 +1364,32 @@ To be removed in 0.20
...
    grid_search.GridSearchCV
+   grid_search.RandomizedSearchCV
+   cross_validation.LeaveOneOut
+   cross_validation.LeavePOut
+   cross_validation.KFold
+   cross_validation.StratifiedKFold
+   cross_validation.ShuffleSplit
+   cross_validation.StratifiedShuffleSplit
+   cross_validation.PredefinedSplit
+   lda.LDA
```
Comment (jnothman) [inline doc/modules/classes.rst]: Yes, as long as these are just import aliases, we can't include them here. Our options are to:
1. remove them from here and leave it at that.
2. do more than just import with rename: make LDA and QDA classes defined as `class LDA(LinearDiscriminantAnalysis): pass` and similar, but there may be other consequences of this.

**Comment by jnothman** [comment]
I'm going to be away for much of the next week. It may be better for you to
make your changes and open a PR to the main project, as you might get good
feedback from another reviewer.

File: doc/modules/classes.rst (hunk 10 above)
Comment (waterponey) [inline doc/modules/classes.rst]: I'll try number 2

**Comment by raghavrv** [comment]
@waterponey Like Joel suggests do you want to make a PR at the main repo for a deeper feedback? I have some time to review this...

**Comment by waterponey** [comment]
Alright

**Comment by waterponey** [comment]
I tried to do that yesterday but interned was broken

**Comment by raghavrv** [comment]
@waterponey Could you close this one? :)
